lib/fluent/plugin/out_http.rb

```diff
@@ -117,7 +119,7 @@ def shutdown
   end
 
   def format_url(tag, time, record)
-    @endpoint_url
+    @current_chunk_endpoint || @endpoint_url
```

Comment:
see if we can remove the logic from the `write` function and just check for `record.timekey` here (maybe under metadata) and eliminate the logic and instance variable from the `write` function.